Security/Authentication/Token/OAuth2Token.php

```diff
@@ -36,10 +37,11 @@ public function getCredentials()
 
     private function buildRolesFromScopes(): array
     {
+        $prefix = $this->getAttribute('role_prefix');
         $roles = [];
 
         foreach ($this->getAttribute('server_request')->getAttribute('oauth_scopes', []) as $scope) {
-            $roles[] = sprintf('ROLE_OAUTH2_%s', trim(strtoupper($scope)));
+            $roles[] = trim(strtoupper($prefix . $scope));
```

Comment:
@Allypost @spideyfusion Should we ensure that `$prefix` ends with `_`?

Reply:
@Allypost @spideyfusion Also, we should probably trim just the `$prefix` variable.

Reply:
@HypeMC I think the prefix should be freely user-definable (eg. if someone wants to separate their roles with a `:` they should be able to)

Reply:
@HypeMC I think the trimming bit is a bit of a micro optimisation (and IMHO it looks slightly cleaner this way)